File: Project_Euler/3.rb
require 'prime'
(600851475143/2).downto(1) do |x|
    if ((600851475143%x)==0)&&(x.prime?)==true
        puts x
        break
    end
end

#(600851475143/2).downto(1){|x| begin puts x; break; end if ((600851475143%x)==0)&&(x.prime?==true)}

File: Project_Euler/3.cpp
#include<iostream>
#include<stdio.h>
using namespace std;
bool prime(long double x)
{
  long double i=2; 
  for(; i<=x/2; i++)
  {
    int res=static_cast<int>(x/i); 
    float mod=x-static_cast<float>(res)*i;
    if (mod==0)
      return false;
  }
  return true;

}

int main()
{
  long double x=600851475143, i;
  for(i=x/2; i>=1; i--)
  {
    int res=static_cast<int>(x/i); 
    float mod=x-static_cast<float>(res)*i;
    if((prime(i)==true)&&(mod==0))
    {
      cout<<i;
      break;
    }
  }
  return 0;
}
